### ISSUE TITLE
Feature Request: Animated GIF wallpaper support

### ISSUE THREAD

**Issue by d3nwah** [cvzi/darkmodewallpaper#113]
Hello :)

As it is in the current F-Droid "1.4.11 (43)" release, you can load a GIF image as a wallpaper, but sadly it doesn't animate. Is it easy to implement? If it's viable I would like to request it :slightly_smiling_face:

Br

**Comment by cvzi** [comment]
I am not sure if it is possible, the app might be too slow for a smooth animation. I will check when I have some time.

**Comment by cvzi** [comment]
I did a quick test and it seems to work smoothly on my phone.

Did you already have specific GIF wallpapers in mind? If so, could you upload them here or send them to me (cuzi-android@openmail.cc)? I would try to create a test .apk with the GIFs included.
That way you could also test if it works on your phone.

The whole graphical interface of the app is currently made for handling static images (i.e. bitmaps), so it will be a lot of work to add support for selecting a GIF from the gallery. I don't want to do that before I am sure it really works smoothly. Adding the GIF directly to the .apk for testing should be quick and easy.

**Comment by d3nwah** [comment]
Of course! The image I use as wallpaper is this lovely pixel art found on web
![09ef55d913c4e979603e468d17e8730a](https://user-images.githubusercontent.com/6751254/218602787-487e83b3-ff18-4ee0-8ebc-85d436f06d04.gif)

Thank you for considering implementing the feature

**Comment by cvzi** [comment]
You can download the apk from [here](https://github.com/cvzi/darkmodewallpaper/releases/tag/v1.4.13): 
https://github.com/cvzi/darkmodewallpaper/releases/download/v1.4.13/gif-debug-1.apk (13.2 MB)

The little previews are not animated, but when you apply the live wallpaper it should be animated.

**Comment by d3nwah** [comment]
Animates like a charm in home screen.

Additional feedback:
- Assuming a scenario without a separate lock screen wallpaper, on the lock screen it displays the properly animated wallpaper but with this new build it's always centered horizontally, whereas before the animated support it remembered what position the wallpaper was at and displayed it (making unlocking more seamless looking)
- In a scenario where both lock screen and home screen use animated wallpapers, after unlocking the performance gets choppy for a second or so (noticeable when swiping to change screens during that time) - probably becasue of the transition and/or the fact it's a debug build. I personally don't intend to use a separate and/or animated lock screen wallpaper, so it's not an issue for me :)
- In a scenario where only home screen uses the wallpaper, without a separate lock screen one, sometimes after unlocking the phone the performance gets a tiny bit stuttery for a sec, but not as bad as in the previous bullet point. Again, probably because it's a debug build

**Comment by cvzi** [comment]
Thanks for the details. I will add it properly then

**Comment by cvzi** [comment]
Here's another version for testing:
https://github.com/cvzi/darkmodewallpaper/releases/download/v1.4.13/gif-debug-2.apk

It allows to select .gif animations (or .webp) from the gallery. Your first point should also be fixed.

**Comment by cvzi** [comment]
That's good 😂
I did improve the performance a little bit compared to the first version. Still, I was able to reproduce the stuttering after unlock on my phone a few times as well, especially with big GIF files. But I don't see a way to improve that further.

I wonder how it affects battery life though, but that's hard to measure. And I never used a animated wallpaper before, so I don't know how they usually affect battery life.

**Comment by cvzi** [comment]
It is now released in the normal version on F-Droid and Google Play.

I looked at the battery usage over 48h on my phone: In the debug version, with your GIF on home-screen and another GIF on the lock-screen. The app now appears in my Android settings in the battery usage list. It is listed as `1%`  *Used for 6 hours*. With static images it never even appeared in the list. So definitely uses more battery, but still seems acceptable.
